pkg/api/types.go

```diff
@@ -719,11 +719,11 @@ type Probe struct {
 	// The action taken to determine the health of a container
 	Handler `json:",inline"`
 	// Length of time before health checking is activated.  In seconds.
-	InitialDelaySeconds int64 `json:"initialDelaySeconds,omitempty"`
+	InitialDelaySeconds int `json:"initialDelaySeconds,omitempty"`
```

Comment:
If teh conversions handle it, fine.  for internal it is a lot less terrible.

On Tue, Nov 17, 2015 at 10:43 PM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/17353#discussion_r45164702>
> :
>
> > @@ -719,11 +719,11 @@ type Probe struct {
> >  	// The action taken to determine the health of a container
> >  	Handler `json:",inline"`
> >  	// Length of time before health checking is activated.  In seconds.
> > -	InitialDelaySeconds int64 `json:"initialDelaySeconds,omitempty"`
> > +	InitialDelaySeconds int `json:"initialDelaySeconds,omitempty"`
>
> Yes, and generated conversions now properly tolerate the change. Changing
> Int internally rippled out to most code paths - if we document int64
> external requires int64 internal (until we nuke internal), we can attack it
> in pieces. On Nov 17, 2015, at 11:29 PM, Tim Hockin <
> notifications@github.com> wrote: In pkg/api/types.go <#17353 (comment)
> <https://github.com/kubernetes/kubernetes/pull/17353#discussion_r45158886>
> >:
> @@ -719,11 +719,11 @@ type Probe struct { // The action taken to determine
> the health of a container Handler `json:",inline"` // Length of time before
> health checking is activated. In seconds. - InitialDelaySeconds int64
> `json:"initialDelaySeconds,omitempty"` + InitialDelaySeconds int
> `json:"initialDelaySeconds,omitempty"`
> We should document that in the conventions. It's obviously easier to use
> int internally. Will the codegen converters convert int32 -> int ? — Reply
> to this email directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/17353/files#r45158886>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17353/files#r45164702>.
>